salt/master.py

```diff
@@ -1269,9 +1269,12 @@ def _file_recv(self, load):
             return {}
         load.pop('tok')
 
+        # Join path
+        sep_path = os.sep.join(load['path'])
```

Comment:
But not all minions will be updated at the same time as the master. We need to handle both cases.